tests/sqlparser_snowflake.rs

```diff
@@ -2974,3 +2974,10 @@ fn test_table_sample() {
     snowflake_and_generic().verified_stmt("SELECT id FROM mytable TABLESAMPLE (10) REPEATABLE (1)");
     snowflake_and_generic().verified_stmt("SELECT id FROM mytable TABLESAMPLE (10) SEED (1)");
 }
+
+#[test]
+fn parse_update_from_before_select() {
```

Comment:
note if we drop the dialect guard, we can move the tests to sqlparser_common.rs to cover all dialects.

Also could we add a test case demonstrating the behavior mentioned above where we have two `FROM` clauses? e.g.
```sql
UPDATE t1 FROM (SELECT ... ) AS t2 SET name = t2.name FROM (SELECT ...) AS t2;
```